paddle/contrib/inference/paddle_inference_api.h

```diff
@@ -64,6 +73,18 @@ class Predictor {
       kAutoMixedTensorRT,  // Automatically mix Fluid with TensorRT.
     };
   };
+
+private:
+  // Build the network before inference.
+  virtual bool Init(const Config& config) = 0;
 };
 
-}  // namespace paddle
+// A factory to help create difference predictor.
+std::unique_ptr<PaddlePredictor> CreatePaddlePredictor(
+    const PaddlePredictor::Config& config);
+
+// Do not use this, just a demo indicating how to customize a config for a
+// specific predictor.
+struct DemoConfig : public PaddlePredictor::Config {
```

Comment:
Let's be more serious. DemoXXX shouldn't show up in api header file... move it to test.cc?